util/terraform.go

```diff
@@ -13,20 +16,56 @@ import (
 	"github.com/coveooss/gotemplate/v3/collections"
 	"github.com/coveooss/gotemplate/v3/json"
 	"github.com/coveooss/gotemplate/v3/template"
+	"github.com/coveooss/gotemplate/v3/utils"
 	"github.com/coveooss/gotemplate/v3/yaml"
 	"github.com/coveooss/multilogger/errors"
 	"github.com/sirupsen/logrus"
 )
 
+type (
+	cachedDefaultVariables struct {
+		values map[string]interface{}
+		all    map[string]*configs.Variable
+		err    error
+	}
+)
+
+var (
+	reTfVariables = regexp.MustCompile(`(?ms)^variable\s+".*?"\s+{(\s*\n.*?^}|\s*}$)`)
+	cacheDefault  = make(map[string]*cachedDefaultVariables)
+)
+
 // LoadDefaultValues returns a map of the variables defined in the tfvars file
 func LoadDefaultValues(folder string) (importedVariables map[string]interface{}, allVariables map[string]*configs.Variable, err error) {
+	var tmpDir string
+	defer func() {
+		if tmpDir != "" {
+			os.RemoveAll(tmpDir)
+		}
+		if cached := cacheDefault[folder]; cached == nil {
+			// We put the result in cache for the next call
+			cacheDefault[folder] = &cachedDefaultVariables{importedVariables, allVariables, err}
+		}
+	}()
+
+	if cached := cacheDefault[folder]; cached != nil {
+		// If we already processed this folder, we simply return the cached values
+		return cached.values, cached.all, nil
+	}
+
+	if tmpDir, err = createTerraformVariablesTemporaryFolder(folder); tmpDir == "" || err != nil {
+		return
+	}
+
 	var terraformConfig *configs.Module
 	parser := configs.NewParser(nil)
-	if terraformConfig, err = parser.LoadConfigDir(folder); err != nil && err.(hcl.Diagnostics).HasErrors() {
-		err = fmt.Errorf("caught errors while trying to load default variable values from %s:\n%w", folder, errors.Array(err.(hcl.Diagnostics).Errs()))
-		return
+	if terraformConfig, err = parser.LoadConfigDir(tmpDir); err != nil {
```

Comment:
Shouldn't we try to load the files normally first and then do this? Creating a temporary directory and using regex is more error prone and it takes more time than having TF compliant configs. I would also log that we failed to load the files

Reply:
I don't think so. Processing all terraform files to read variables is also time consuming. But I can do it if you prefer.

Reply:
It probably fails really quickly if the files aren't valid

Reply:
And I just think it's unexpected behavior. By trying first, logging if it fails and retrying with the copy, it's more explicit and expected